satpy/writers/oracle.py

```diff
@@ -0,0 +1,102 @@
+"""Module for writing feature datasets to Oracle databases."""
+import oracledb
+import shapely
+
+from satpy.writers import Writer
+
+
+class oracle(Writer):
+    """Write features in geodataframe to Oracle database."""
+    def __init__(self, name=None, filename=None, base_dir=None, layer_name=None, **kwargs):
+        """Init."""
+        super().__init__(name, filename, base_dir,  **kwargs)
+
+    def save_dataset(self, dataset, table_name=None, user=None, password=None,
+                     host=None, port=None, service_name=None, **kwargs):
+        """Write data to Oracle database table."""
+        write_gdf_to_oracle(dataset.data, table_name, user, password, host, port, service_name)
+
+def create_multiline_geometry(multilines, t_obj, einfo_obj, otype_obj):
+    """Creates an Oracle SDO_GEOMETRY object for a MultiLineString."""
+    geometry = t_obj.newobject()
+    geometry.SDO_GTYPE = 2006  # 2006 = MULTILINESTRING
+    geometry.SDO_SRID = 4326   # SRID for WGS 84
+    geometry.SDO_ELEM_INFO = einfo_obj.newobject()
+    geometry.SDO_ORDINATES = otype_obj.newobject()
+
+    elem_info = []
+    ordinates = []
+    start_index = 1
+
+    for line in multilines.geoms:
+        line_coords = [coord for point in line.coords for coord in point]
+        elem_info.extend([start_index, 2, 1])  # Start index, LineString, Simple Element
+        ordinates.extend(line_coords)
+        start_index += len(line_coords)  # Update start index for next line
+
+    geometry.SDO_ELEM_INFO.extend(elem_info)
+    geometry.SDO_ORDINATES.extend(ordinates)
+
+    return geometry
+
+def write_gdf_to_oracle(gdf, table_name, user, password, host, port, service_name):
+    """Write geodataframe to oracle database.
+
+    The connection data like user/pw/host/port/service_name are taken from the config.
```

Comment:
I don't see that they are taken from the config, rather they are passed by the user to this function.